tensorcircuit/timeevol.py

```diff
@@ -427,37 +427,56 @@ def _evol(t: Tensor) -> Tensor:
 ed_evol = hamiltonian_evol
 
 
-@partial(arg_alias, alias_dict={"h_fun": ["hamiltonian"], "t": ["times"]})
-def evol_local(
-    c: Circuit,
-    index: Sequence[int],
-    h_fun: Callable[..., Tensor],
-    t: float,
-    *args: Any,
-    **solver_kws: Any,
-) -> Circuit:
-    """
-    ode evolution of time dependent Hamiltonian on circuit of given indices
-    [only jax backend support for now]
+def _solve_ode(
+    f: Callable[..., Tensor],
+    s: Tensor,
+    times: Tensor,
+    args: Sequence[float | int | complex],
+    solver_kws: Dict[str, Any],
+) -> Tensor:
+    rtol = solver_kws.get("rtol", 1e-12)
+    atol = solver_kws.get("atol", 1e-12)
 
-    :param c: _description_
-    :type c: Circuit
-    :param index: qubit sites to evolve
-    :type index: Sequence[int]
-    :param h_fun: h_fun should return a dense Hamiltonian matrix
-        with input arguments time and *args
-    :type h_fun: Callable[..., Tensor]
-    :param t: evolution time
-    :type t: float
-    :return: _description_
-    :rtype: Circuit
-    """
-    s = c.state()
-    n = int(np.log2(s.shape[-1]) + 1e-7)
-    if isinstance(t, float):
-        t = backend.stack([0.0, t])
-    s1 = ode_evol_local(h_fun, s, t, index, None, *args, **solver_kws)
-    return type(c)(n, inputs=s1[-1])
+    ts = backend.convert_to_tensor(times)
+    ts = backend.cast(ts, dtype=rdtypestr)
+
+    max_steps = solver_kws.get("max_steps", 10000)
+
+    if (solver := solver_kws.get("solver", "Dopri5")) == "Dopri5":
```

Comment:
why dopri5 corresponds the original odeint in jax, doesn't diffrax provides dopri5?